app/models/user.rb

```diff
@@ -3,13 +3,19 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable, :confirmable
-  has_many :posts, foreign_key: 'author_id'
-  has_many :comments, foreign_key: 'author_id'
-  has_many :likes, foreign_key: 'author_id'
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, foreign_key: 'author_id', dependent: :destroy
+  has_many :likes, foreign_key: 'author_id', dependent: :destroy
   validates :name, presence: true
   validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
+  # Roles = [:admin, :default]
+
   def recent_post
     posts.order('created_at Desc').limit(3)
   end
+
+  def is?(requested_role)
+    role == requested_role.to_s
+  end
 end
```

Comment:
- [ ] [OPTIONAL] It would be better to create `user.admin?` method to check if the user is admin or not. The code would read
```
   def admin?
     role == 'admin' ? true : false
   end